python_modules/jupedsim/jupedsim/distributions.py

```diff
@@ -272,39 +274,68 @@ def distribute_in_circles_by_density(*, polygon, distance_to_agents, distance_to
                                            numbers_of_agents=number_of_agents, seed=seed, max_iterations=max_iterations)
 
 
-def distribute_til_full(*, polygon, distance_to_agents, distance_to_polygon, seed=None, max_iterations=10000):
-    """returns as many randomly placed points as fit into the polygon
+def distribute_till_full(*, polygon, distance_to_agents, distance_to_polygon, seed=None, max_iterations=10_000, k=30):
+    """returns as many randomly placed points as fit into the polygon. 
+    Points are distributed using Bridson’s algorithm for Poisson-disc sampling
+    The algorithm is explained in Robert Bridson´s Paper "Fast Poisson Disk Sampling in Arbitrary Dimensions"
         :param polygon: shapely polygon in which the agents will be placed
         :param distance_to_agents: minimal distance between the centers of agents
         :param distance_to_polygon: minimal distance between the center of agents and the polygon edges
         :param seed: define a seed for random generation, Default value is None which corresponds to a random value
         :param max_iterations: no more than max_iterations must find a point inside the polygon, default is 10_000
+        :param k: around each point k point will be created before the point is considered inactive
         :return: list of created points"""
     if not isinstance(polygon, shply.polygon.Polygon):
         raise IncorrectParameterError(f"Polygon is expected to be a shapely Polygon")
     box = __get_bounding_box(polygon)
 
     np.random.seed(seed)
-
+    # initialises a list for active Points and a Grid administering all created points
+    active = []
     grid = Grid(box, distance_to_agents)
-    iterations = 0
-    while True:
-        if iterations > max_iterations:
+    # initialisation of the first point
+    iteration = 0
+    while iteration < max_iterations:
+        first_point = (np.random.uniform(box[0][0], box[1][0]), np.random.uniform(box[0][1], box[1][1]))
+        if __check_distance_constraints(first_point, distance_to_polygon, grid, polygon):
+            grid.append_point(first_point)
+            active.append(first_point)
             break
-        temp_point = (np.random.uniform(box[0][0], box[1][0]), np.random.uniform(box[0][1], box[1][1]))
-        if __check_distance_constraints(temp_point, distance_to_polygon, grid, polygon):
-            grid.append_point(temp_point)
-
-            iterations = 0
-        else:
-            iterations += 1
+        iteration = iteration + 1
+    if iteration >= max_iterations:
+        raise IncorrectParameterError("The first point could not be placed inside the polygon."
+                                      " Check if there is enough space for agents provided inside the polygon")
+
+    # while points are active a random reference point is selected
+    while active:
+        ref_point = active[np.random.randint(0, len(active))]
```

Comment:
Shouldn't it be: `np.random.randint(0, len(active)-1)` ?

Reply:
no. 
random.randint(low, high=None, ... )
Return random integers from low (inclusive) to high (exclusive).